client/src/components/NewBlockForm.tsx

```diff
@@ -65,8 +65,11 @@ export class NewBlockFormPresentational extends React.Component<any, any> {
     };
   }
 
-  public componentWillReceiveProps() {
-    if (this.state.pending) {
+  public componentWillReceiveProps(newProps) {
+    if (
+      this.state.pending &&
+      newProps.numBlocksInRedux > this.props.numBlocksInRedux // This checks whether the newly asked sub-question has made it into Redux
```

Comment:
This seems fine as long as Redux will trigger a new props refresh, which it should.

Do you think the `setState` on 81 should be moved out of this block?